### DISCUSSION TITLE
is MongoDB extension not supported?

### DISCUSSION THREAD

**Discussion by bornemisza** [php/frankenphp#584] [Q&A]
I ran into similar problem like someone else as well (https://github.com/mongodb/laravel-mongodb/issues/2701), that running Laravel Octane with FrankenPHP. Seems like MongoDB extension does not work with FrankenPHP. I have tested with Roadrunner and Swole also, but it worked perfectly fine with them.

Reproduction:
- Extend Laravel Sail dockerfile with "pecl install mongodb" and add the "extension=mongodb.so" to php.ini. (PHP 8.1 - 8.3)
- Install "mongodb/laravel-mongodb" and follow this guide: https://www.mongodb.com/docs/drivers/php/laravel-mongodb/current/quick-start/configure-mongodb/
- Try to store something into the database and you'll recieve `Class "MongoDB\Driver\Manager" not found.`.

I have checked with `php -m` and `phpinfo()` that mongodb extension is enabled/activated.

I have not found any related docs or code to solve this issue. Maybe it'll be a pinpoint for someone as well. Any thought from community?

**Comment by dunglas** [comment]
You need to use Octane with Docker for now to use MongoDB. The static binary doesn't support Mongo yet. We could add support for Mongo for the static binary on Linux, but not for macOS yet: https://static-php.dev/en/guide/extensions.html

**Reply by bornemisza** [reply]
Thank you @dunglas for your quick reply. I'm currently using Octane with Docker especially running on Ubuntu 22.04, so I would appreciate to have it on Linux. Possible to get a notification somehow if you add support for it?

**Reply by dunglas** [reply]
If you are using Docker this should already work. Just install the Mongo extension with the provided script.

**Reply by bornemisza** [reply]
Interesting. It is installed still not working, like I said in main discussion. What do you think where do I go sideway? Happy to debug more btw. Should I create a repo to demo?

**Reply by withinboredom** [reply]
Try running `frankenphp php-cli` and checking the output of phpinfo for the mongo extension. This way, you can be sure that whatever `php.ini` you have frankenphp including is showing the extension as being installed.

Also, I recommend using `install-php-extensions mongodb` instead of using pecl directly, as it [enables a few settings for you that are probably recommended](https://github.com/mlocati/docker-php-extension-installer/blob/615bfd2522ae7ceaccf55d66cc14345fc6850358/install-php-extensions#L3152)

**Comment by hugohenrique** [comment]
I'm also going through the same problem.
I have already confirmed that the extension is installed and configured.
![Screenshot from 2024-03-01 19-09-12](https://github.com/dunglas/frankenphp/assets/4316743/29556127-26ac-4e3c-82c1-f32b60255bb4)

**Comment by hugohenrique** [comment]
Does anyone have any idea what it could be?

**Comment by huangjian80s** [comment]
Thank you very much. I just bought MongoDB and want to use it as soon as possible.

I searched for a long time and finally found the reason here

Can you provide a static binary for Linux with MongoDB support? I'm using PHP version 8.4.
